lib/puppet/provider/certmonger_certificate/certmonger_certificate.rb

```diff
@@ -0,0 +1,149 @@
+Puppet::Type.type(:certmonger_certificate).provide :certmonger_certificate do
+  desc "Provider for certmonger certificates."
+
+  confine :exists => "/usr/sbin/certmonger"
+  commands :getcert => "/bin/ipa-getcert"
```

Comment:
Using ipa-getcert will limit this to only working with IPA-issued certificates. It is more general-purpose to use getcert.